Tools/Scripts/webkitpy/benchmark_runner/benchmark_runner.py

```diff
@@ -85,6 +93,52 @@ def available_plans():
         plans = [os.path.splitext(plan_file)[0] for plan_file in os.listdir(BenchmarkRunner.plan_directory()) if plan_file.endswith(".plan")]
         return plans
 
+    @staticmethod
+    def format_subtests(subtests):
+        # An ordered dictionary helps us prioritize listing subtests without a suite (denoted with key: "")
+        subtests = OrderedDict(subtests)
+        subtest_list = []
+        for suite, tests in subtests.items():
+            subtest_list += ['{suite}/{test}'.format(suite=suite, test=test) if suite else test for test in tests]
+        return subtest_list
+
+    @staticmethod
+    def available_subtests(plan_file):
+        plan = BenchmarkRunner._load_plan_data(plan_file)[1]
+        return plan.get('subtests', {})
+
+    def validate_subtests(self, subtests):
+        if 'subtests' not in self._plan:
+            raise Exception('Subtests are not available for the specified plan.')
+        valid_subtests = OrderedDict(self._plan['subtests'])
+        subtests_to_run = defaultdict(list)
+        for subtest in subtests:
+            if '/' in subtest:
+                subtest_suite, subtest_name = subtest.split('/')
+            else:
+                subtest_suite, subtest_name = None, subtest
+            did_append_subtest = False
+            if subtest_suite:
+                if subtest_suite not in valid_subtests:
+                    _log.warning('{} does not belong to a valid suite, skipping'.format(subtest))
+                    continue
+                if subtest_name in valid_subtests[subtest_suite]:
+                    subtests_to_run[subtest_suite].append(subtest_name)
+                    did_append_subtest = True
+            else:  # If no suite is provided, we should do our best to infer one.
+                for suite, tests in valid_subtests.items():
+                    if subtest_name in tests:
+                        subtests_to_run[suite].append(subtest_name)
+                        did_append_subtest = True
+                        break
```

Comment:
Does this mean we will use the first match if there are multiple tests with same name but under different suites?

Reply:
Yep - first match alphabetically ordered by suite (or no suite, if one exists)